Pydantic alias

auerj opened the pull request:
Introduces aliases to:
- make reserved keyword (e.g. class) member accessible
- changed the pydantic model to snake_case with camelCase aliases

This allows us to enable unittests for the generated r4 code

Fixes #6

Commit:
snake case pydantic properties with alias

Commit:
added alias tests

Commit:
updated gitignore

auerj committed:
- Incooperated some extensions suggested by daniel
- added some tests
- disabled tests until Resource deserialization issues are solved

auerj committed:
finalized snake case members

julian-r commented on fhirzeug/generators/python_pydantic/templates/codesystems.py.jinja2:
```diff
@@ -1,6 +1,9 @@
 
-class {{system.name}}(enum.Enum):
+class {{system.name}}(str, enum.Enum):
```
Don't get this, doc needed.

julian-r commented on fhirzeug/fhirclass.py:
```diff
@@ -194,21 +195,43 @@ def resource_type_enum(self):
     @property
     def choice_properties(self) -> Dict[str, list]:
+        return self._choice_properties()
+
```
lets use a jinja filter

julian-r committed:
snake case with template only solution

julian-r committed:
using stringcase for snakecase generation

julian-r committed:
moved model config to superclass

julian-r committed:
removed _without_empty_items, so we can put that in a seperate pr

julian-r reviewed (approved):
Made some changes,

- Moved the model-config into FHIRAbstractBase
- Snake case is now done with a tempalte filter, IMHO this is a more abstract future language approve way of doing it.

julian-r committed:
moved baseclass

julian-r committed:
reverted codsystem template

cannot break tests, with uncommenting the str part, but will now keep it in mind

julian-r committed:
added missing whitespace
